OneId System : initial release

Akiyamka opened the pull request:
<!--
Thank you for your pull request! 

Please title your pull request like this: 'Module: Change', eg 'Fraggles Bid Adapter: support fragglerock'

Please make sure this PR is scoped to one change or you may be asked to resubmit. 
 
Please make sure any added or changed code includes tests with greater than 80% code coverage. 

See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.

For any user facing change, submit a link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: also submit your bidder parameter documentation as noted in https://docs.prebid.org/dev-docs/bidder-adaptor.html#submitting-your-adapter -->
- [ ] Updated bidder adapter  <!--  IMPORTANT: (1) consider whether you need to upgrade your bidder parameter documentation in https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders and (2) if you have a Prebid Server adapter, please consider whether that should be updated as well. --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
New Id system module for [OneId](https://docs.oneid.xyz/developers-guide/oneid-sdk) 
Contact email of maintainer: alexandr.dubinin@audienzz.com

## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

jlquaccia commented on modules/oneIdIdSystem.js:
```diff
@@ -0,0 +1,172 @@
+   * @param {SubmoduleConfig} [config]
+   * @returns {IdResponse|undefined}
+   */
+  getId(config) {
```
do you need to provide any validations against user consentData?

jlquaccia replied:
That makes sense.  If you guys already have something in place, sounds like you are already good there then.

Looks like there is a test failing within the circleci e2e test.. otherwise, everything else LGTM!

jlquaccia commented:
@Akiyamka left one comment.  Also noticed that one circle ci test is failing

Akiyamka commented:
@jlquaccia 
I apologize for the delay. I encountered an issue where I couldn't run the tests locally. It turned out that the current tests were using a legacy API - switchToFrame (https://webdriver.io/docs/api/webdriver/#switchtoframe). I updated the WebDriver version and modified the test utilities to use the new API (switchToFrame) - now all tests pass, including the one that was previously failing in CI. .

patmmccann commented:
Reverted for rules violation; tell us more about this script you are injecting

Akiyamka commented:
@patmmccann I would be very grateful if you could provide a bit more information for context - which specific rules were violated?

This script loads the SDK for interacting with the [https://www.oneid.digital/](https://www.oneid.digital/) system.

patmmccann commented:
Please do not load the sdk. We have four exceptions for id modules, and each one is a discussion of sorts with only id5 in a endstate we're pretty happy with, https://github.com/prebid/Prebid.js/blob/b7b5cfac4dec925e502467afa5bd6164b17668d8/src/adloader.js#L40, but we'd prefer not to add more without compelling reasoning. Also if you do end up loading one, you cnnot use appendchild to do so, you have to use the methods in that linked library

Akiyamka commented:
@patmmccann, thank you for the explanation. I apologize for violating your rules; it was not intentional.

In theory, I can make this code work without using third-party scripts through reverse engineering (I already have such a version). However, this solution does not appear reliable, as the contracts used within the SDK for API communication may change at any time.

The source code of this SDK is not published anywhere and is not distributed through any package managers that would allow for gradual migration in case of changes. Therefore, the approach of loading the ready-made SDK was chosen as the only reliable and officially supported option for using this identification system.

What do you think about this? Is this sufficient justification to allow the use of this SDK?
